core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -273,8 +273,13 @@ object FileUtils {
   @JvmStatic
   fun getSdCardMainPath(context: Context, storageName: String) =
     context.getExternalFilesDirs("")
-      .first { it.path.contains(storageName) }
-      .path.substringBefore(context.getString(R.string.android_directory_seperator))
+      .firstOrNull { it.path.contains(storageName) }
+      ?.path?.substringBefore(context.getString(R.string.android_directory_seperator))
+      // The `context.getExternalFilesDirs("")` method does not provide the path for USB sticks due to security changes,
+      // but it does provide the path for the 'SD Card'.
+      // Therefore, the code below will only work when the user selects the
+      // ZIM file from a USB stick. USB sticks are mounted under the `/mnt/media_rw` directory.
+      ?: "/mnt/media_rw/$storageName"
```

Comment:
@MohitMaliDeveloper This confuses the functionality of the method. Maybe in the upper level, we should handle the path?

Reply:
@gouri-panda Good catch! I have improved the method and comments.